nova/scheduler/weights/affinity.py

```diff
@@ -49,7 +56,26 @@ def _weigh_object(self, host_state, request_spec):
         members = set(request_spec.instance_group.members)
         member_on_host = instances.intersection(members)
 
-        return len(member_on_host)
+        weight = len(member_on_host)
+        # if the host contained servers from the same group, return that weight
+        if weight:
+            return weight
+
+        # get shards of the host
+        host_shard_aggrs = [aggr for aggr in host_state.aggregates
+                            if aggr.name.startswith(self._SHARD_PREFIX)]
```

Comment:
This should really just be one. If it's more: print a warning and take the first.

Reply:
done